pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -1120,6 +1127,74 @@ func (r ReconcileHyperConverged) amendBadDefaults(req *common.HcoRequest) (bool,
 	return modified, nil
 }
 
+func (r ReconcileHyperConverged) applyUpgradePatches(req *common.HcoRequest) (bool, error) {
+	modified := false
+
+	knownHcoVersion, _ := req.Instance.Status.GetVersion(hcoVersionName)
+	if knownHcoVersion == "" {
+		knownHcoVersion = "0.0.0"
+	}
+
+	err := readUpgradePatchesFromFile(req)
+	if err != nil {
+		return false, err
+	}
+	err = validateUpgradePatches(req)
+	if err != nil {
+		return false, err
+	}
```

Comment:
Can we do it only once, when getting into upgrade mode, and cache the result?

Reply:
done